dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/Category/Save/SaveCategoryDifferentScopesTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Catalog\Controller\Adminhtml\Category\Save;
+
+use Magento\Catalog\Api\CategoryRepositoryInterface;
+use Magento\Catalog\Api\Data\CategoryInterface;
+use Magento\Catalog\Model\Category;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * @magentoAppArea adminhtml
+ * @magentoDbIsolation disabled
+ */
+class SaveCategoryDifferentScopesTest extends AbstractSaveCategoryTest
+{
+    private const DEFAULT_STORE_ID = 1;
+    private const FIXTURE_CATEGORY_ID = 333;
+    private const GLOBAL_SCOPE_ID = 0;
+
+    private const FIXTURE_URL_KEY = 'category-1';
+
+    private const FIXTURE_CATEGORY_ROOT = 400;
+    private const FIXTURE_CATEGORY_LEAF = 402;
+
+    private const FIXTURE_SECOND_STORE_CODE = 'fixturestore';
+
+    /** @var CategoryRepositoryInterface */
+    private $categoryRepository;
+
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->categoryRepository = $this->_objectManager->get(CategoryRepositoryInterface::class);
+        $this->storeManager = $this->_objectManager->get(StoreManagerInterface::class);
+    }
+
+    /**
+     * @magentoDataFixture Magento/Catalog/_files/category.php
+     */
+    public function testUpdateUrlKeyForStoreExpectNoChangeOfGlobalUrlPath()
```

Comment:
Is this one supposed to be failing for the current core functionality?

Reply:
That is the tricky part. This should not fail, but I also noticed it is failing, that means that in the last 3 months, that feature was broken. I'm going to investigate and fix this in the scope of this PR.

Reply:
@lbajsarowicz do you had a chance to investigate the issue?

Reply:
Let me merge the latest `2.4-develop` branch and run that once again.
I'm pretty sure I know which exactly PR broke that, but I'll handle that at the beginning of September unless you arrange someone else.